pkg/knode-manager/controllers/pvc.go

```diff
@@ -0,0 +1,250 @@
+	pvc2key, result := p.downstreamQueue.Get()
+	if result {
+		klog.Errorf("kosmos sync downstream: get pvc2key from downstream queue failed")
+		return
+	}
+	defer p.downstreamQueue.Done(pvc2key)
+	pvcNs, pvcName, err := cache.SplitMetaNamespaceKey(pvc2key.(string))
+	if err != nil {
+		p.downstreamQueue.Forget(pvc2key.(string))
+		return
+	}
+	klog.Infof("kosmos sync downstream: start sync downstream pvc, pvc: %v", pvc2key.(string))
+	pvcInstance, err := p.handler.Get(context.TODO(), pvcNs, pvcName)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			klog.Errorf("kosmos sync downstream: get pvc from downstream list failed, err: %v", err)
+			return
+		}
+		klog.Warningf("kosmos sync downstream: pvc has been deleted from downstream list, pvc: %v", pvcName)
+	}
+
+	klog.Infof("kosmos sync downstream: start sync pvc status, pv: %v", pvcName)
+	key2pvc, err := cache.MetaNamespaceKeyFunc(pvcInstance)
+	if err != nil {
+		runtime.HandleError(err)
+		return
+	}
+	defer func() {
+		if err != nil {
+			klog.Error(err)
+			p.downstreamQueue.AddRateLimited(key2pvc)
+			return
+		}
+	}()
+	pvcInstanceDeepCopy := pvcInstance.DeepCopy()
+	upstreamPVC, err := p.lister.PersistentVolumeClaims(pvcInstance.Namespace).Get(pvcInstance.Name)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			klog.Errorf("kosmos sync downstream: get pvc from upstream list failed, pv: %v", err)
+			return
+		}
+		klog.Warningf("kosmos sync downstream: pvc has been deleted from upstream list, pvc: %v", pvcName)
+	}
+	if err = p.filterPVC(pvcInstanceDeepCopy, p.factor); err != nil {
+		return
+	}
+	_, err = p.patchPVC(upstreamPVC, pvcInstanceDeepCopy, p.client)
+	if err != nil {
+		klog.Errorf("kosmos sync downstream: patch pvc from upstream failed, pv: %v", pvcInstance.Name)
+		return
+	}
+	p.downstreamQueue.Forget(key2pvc)
+	klog.Infof("kosmos sync downstream: finish sync downstream pvc, pvc: %v", pvc2key.(string))
+}
+
+func (p *PVCController) patchPVC(old, new *corev1.PersistentVolumeClaim, c kubernetes.Interface) (*corev1.PersistentVolumeClaim, error) {
+	if reflect.DeepEqual(old.Spec, new.Spec) && reflect.DeepEqual(old.Status, new.Status) {
+		return old, nil
+	}
+
+	oldByte, err := json.Marshal(old)
+	if err != nil {
+		return nil, err
+	}
+	newByte, err := json.Marshal(new)
+	if err != nil {
+		return nil, err
+	}
+	patch, err := jsonpatch.CreateMergePatch(oldByte, newByte)
+	if err != nil {
+		return nil, err
+	}
+
+	newPVC, err := c.CoreV1().PersistentVolumeClaims(old.Namespace).Patch(context.TODO(), old.Name, types.MergePatchType, patch, metav1.PatchOptions{})
+	if err != nil {
+		return old, err
+	}
+	return newPVC, nil
+}
+
+func (p *PVCController) filterPVC(pvc *corev1.PersistentVolumeClaim, factor string) error {
+	labelSelector := pvc.Spec.Selector.DeepCopy()
+	pvc.Spec.Selector = nil
+	utils.TrimObjectMeta(&pvc.ObjectMeta)
+	if pvc.Annotations == nil {
+		pvc.Annotations = map[string]string{}
+	}
+	pvc.Annotations["global"] = "true"
+	if labelSelector != nil {
```

Comment:
`pvc.Annotations["global"] = "true"` 
annotaions added by kosmos would be better with a `kosmos.io/` prefix

Reply:
ok, I changed it